src/modules/menu/components/advancedTools/ImportFile/ImportFileOnlineTab.vue

```diff
@@ -0,0 +1,110 @@
+<script setup>
+import axios, { AxiosError } from 'axios'
+import { computed, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+import ImportFileButtons from '@/modules/menu/components/advancedTools/ImportFile/ImportFileButtons.vue'
+import { handleFileContent } from '@/modules/menu/components/advancedTools/ImportFile/utils'
+import log from '@/utils/logging'
+import { isValidUrl } from '@/utils/utils'
+
+const REQUEST_TIMEOUT = 5 * 60 * 1000 // milliseconds
+
+const i18n = useI18n()
+
+// Reactive data
+const fileUrl = ref('')
+const buttonState = ref('default')
+const errorMessage = ref(null)
+const fileLoaded = ref(false)
+
+// Computed properties
+const isValid = computed(() => !errorMessage.value && fileLoaded.value)
+const isInvalid = computed(() => errorMessage.value)
+
+// Methods
+function onFileUrlChange(event) {
+    errorMessage.value = null
+    fileUrl.value = event.target.value
+}
+
+function validateUrl() {
+    errorMessage.value = null
+    if (!fileUrl.value) {
+        errorMessage.value = 'no_url'
+    } else if (!isValidUrl(fileUrl.value)) {
+        errorMessage.value = 'invalid_url'
+    }
+    return !errorMessage.value
+}
+
+function clearUrl() {
+    fileUrl.value = ''
+    errorMessage.value = null
+}
+
+async function loadFile() {
+    if (!validateUrl()) {
+        return
+    }
+    buttonState.value = 'loading'
+    let response
+
+    try {
+        response = await axios.get(fileUrl.value, { timeout: REQUEST_TIMEOUT })
+        if (response.status !== 200) {
+            throw new Error(`Failed to fetch ${fileUrl.value}; status_code=${response.status}`)
+        }
+        handleFileContent(response.data, fileUrl.value)
+        buttonState.value = 'succeeded'
+        fileLoaded.value = true
+    } catch (error) {
+        buttonState.value = 'default'
+        if (error instanceof AxiosError || /fetch/.test(error.message)) {
+            log.error(`Failed to load file from url ${fileUrl.value}`, error)
+            errorMessage.value = 'loading_error_network_failure'
+        } else {
+            log.error(`Failed to parse file from url ${fileUrl.value}`, error)
+            errorMessage.value = 'invalid_kml_gpx_file_error'
+        }
+    }
+}
+</script>
+
+<template>
+    <!-- Online Tab -->
+    <div id="nav-online" class="tab-pane fade" role="tabpanel" aria-labelledby="nav-online-tab">
+        <form class="input-group d-flex needs-validation">
+            <input
+                type="text"
+                class="form-control import-input"
+                :class="{
+                    'rounded-end': !fileUrl?.length,
+                    'is-valid': isValid,
+                    'is-invalid': isInvalid,
+                }"
+                :placeholder="$t('import_online_placeholder')"
```

Comment:
that sort of work (I don't know how as we've deactivated the legacy mode), but it might better to replace it by `:placeholder="i18n.t('import_online_placeholder')"` so that we may not be suprised by a future update of `vue-i18n` that removes the `$t` support altogether